src/lib/containers/SubsectionRowRenderer.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useState } from 'react'
+import { insertConditionsFromSearchParams, KeyValue, parseEntityIdFromSqlStatement, SQLOperator } from '../utils/functions/sqlFunctions'
+import { SynapseClient, SynapseConstants } from '../utils'
+import {
+  EntityColumnType,
+  QueryBundleRequest,
+  RowSet,
+} from '../utils/synapseTypes/Table'
+import { useSynapseContext } from '../utils/SynapseContext'
+import MarkdownSynapse from './MarkdownSynapse'
+import { SkeletonTable } from '../assets/skeletons/SkeletonTable'
+import { ColumnSpecifiedLink } from './CardContainerLogic'
+
+export type FriendlyValuesMap = {
+  [index: string]: string
+}
+
+export type SubsectionRowRendererProps = {
+  sql: string
+  isMarkdown?: boolean
+  sqlOperator?: SQLOperator
+  searchParams?: KeyValue
+  columnLink?: ColumnSpecifiedLink
+  friendlyValuesMap?: FriendlyValuesMap
+  columnNameIsSectionTitle?: boolean
+  limit?: number
+}
+
+const LIST_COLUMN_TYPES = [EntityColumnType.BOOLEAN_LIST, EntityColumnType.DATE_LIST, EntityColumnType.ENTITYID_LIST, EntityColumnType.INTEGER_LIST, EntityColumnType.STRING_LIST]
+
+const SubsectionRowRenderer: React.FunctionComponent<SubsectionRowRendererProps> = ({
+  sql,
+  searchParams,
+  sqlOperator,
+  isMarkdown = false,
+  columnLink,
+  friendlyValuesMap,
+  columnNameIsSectionTitle = false,
+  limit,
+}) => {
+  const { accessToken } = useSynapseContext()
+  const [rowSet, setRowSet] = useState<RowSet>()
+  const [isLoading, setIsLoading] = useState<boolean>()
+  let mounted = true
+  useEffect(() => {
+    const fetchData = async function () {
+      setIsLoading(true)
+      const sqlUsed = insertConditionsFromSearchParams(
+        sql,
+        searchParams,
+        sqlOperator,
+      )
+      const entityId = parseEntityIdFromSqlStatement(sql)
+      const partMask = SynapseConstants.BUNDLE_MASK_QUERY_RESULTS
+      const request: QueryBundleRequest = {
+        partMask,
+        concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+        entityId,
+        query: {
+          sql: sqlUsed,
+          limit,
+        },
+      }
+
+      const queryResultBundle = await SynapseClient.getFullQueryTableResults(
+        request,
+        accessToken,
+      )
```

Comment:
I didn't realize we had this method and I'm not crazy about it. I'm guessing that we're pretty confident we'll never have to render a huge table in this component.

Reply:
Doesn't need to be used here, so I'll change it!  Useful for those edge cases when we need all of the data before we can show anything (typically plotting)